src/openms/source/CHEMISTRY/EnzymesDB.cpp

```diff
@@ -0,0 +1,340 @@
+          const_enzymes_.insert(enzy_ptr);
+          prefix = split[0] + split[1];
+        }
+        String value = it->value;
+        String key = it.getName();
+        values[key] = value;
+      }
+
+      // add last enzyme
+      enzy_ptr = parseEnzyme_(values);
+      enzymes_.insert(enzy_ptr);
+      const_enzymes_.insert(enzy_ptr);
+    }
+    catch (Exception::BaseException& e)
+    {
+      throw Exception::ParseError(__FILE__, __LINE__, __PRETTY_FUNCTION__, e.what(), "");
+    }
+  }
+
+  void EnzymesDB::clear_()
+  {
+    clearEnzymes_();
+  }
+
+  void EnzymesDB::clearEnzymes_()
+  {
+    set<Enzyme*>::iterator it;
+    for (it = enzymes_.begin(); it != enzymes_.end(); ++it)
+    {
+      delete *it;
+    }
+    enzymes_.clear();
+    enzyme_names_.clear();
+    enzyme_regex_.clear();
+    const_enzymes_.clear();
+  }
+
+  Enzyme* EnzymesDB::parseEnzyme_(Map<String, String>& values)
+  {
+    Enzyme* enzy_ptr = new Enzyme();
+
+    for (Map<String, String>::iterator it = values.begin(); it != values.end(); ++it)
+    {
+      String key(it->first);
+      String value(it->second);
+
+      if (key.hasSuffix(":Name"))
+      {
+        enzy_ptr->setName(value);
+        continue;
+      }
+      if (key.hasSuffix(":RegEx"))
+      {
+        enzy_ptr->setRegEx(value);
+        continue;
+      }
+      if (key.hasSuffix(":RegExDescription"))
+      {
+        enzy_ptr->setRegExDescription(value);
+        continue;
+      }
+      if (key.hasSuffix(":NTermGain"))
+     {
+        enzy_ptr->setNTermGain(EmpiricalFormula(value));
+        continue;
+      }
+      if (key.hasSuffix(":CTermGain"))
+      {
+        enzy_ptr->setCTermGain(EmpiricalFormula(value));
+        continue;
+      }
+      if (key.hasSubstring("PSIid"))
+      {
+        // no PSIid defined?
+        if (!key.hasSuffix(":"))
+        {
+          enzy_ptr->setPSIid(value);
+        }
+        continue;
+      }
+      if (key.hasSubstring("XTANDEMid"))
+      {
+        if (!key.hasSuffix(":"))
+        {
+          enzy_ptr->setXTANDEMid(value);
+        }
+        continue;
+      }
+      if (key.hasSubstring("OMSSAid"))
+      {
+        if (!key.hasSuffix(":"))
+        {
+          enzy_ptr->setOMSSAid(value.toInt());
+        }
+        continue;
+      }
+      if (key.hasSubstring("Synonyms"))
+      {
+        // no synonyms defined?
+        if (!key.hasSuffix(":"))
+        {
+          enzy_ptr->addSynonym(value);
+        }
+        continue;
+      }
+      cerr << "unknown key: " << key << ", with value: " << value << endl;
+    }
+    return enzy_ptr;
+  }
+
+  void EnzymesDB::buildEnzymeNames_()
+  {
+    set<Enzyme*>::iterator it;
+    for (it = enzymes_.begin(); it != enzymes_.end(); ++it)
+    {
+      enzyme_names_[(*it)->getName()] = *it;
+      if ((*it)->getRegEx() != "")
+      {
+        enzyme_regex_[(*it)->getRegEx()] = *it;
+      }
+      set<String>::iterator sit;
+      set<String> syn = (*it)->getSynonyms();
```

Comment:
I know it is done in some other parts of OpenMS but I would not add the synonyms. Better to have one name as global identifier in OpenMS than several that link to the same enzyme.